File: app/src/main/java/com/michalbaran/incomeapp/AddTransactionActivity.java
package com.michalbaran.incomeapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.app.DatePickerDialog;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.text.Layout;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.DatePicker;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.Spinner;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Calendar;

public class AddTransactionActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener {

    TextView txtDate;
    DatePickerDialog.OnDateSetListener dateSetListener;
    Button plusBtn, minusBtn, addBtn;
    LinearLayout layout;
    Context context;
    EditText amountEdit, categoryEdit;
    char activityMode;
    Spinner categorySpinner;
    String categoryTxt;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_transaction);

        Intent intent = getIntent();
        //if (activityMode != null) {
            activityMode = intent.getCharExtra("ACTIVITY_MODE",'0');
        //}
        // activityMode = '0';

        plusBtn = findViewById(R.id.btnPlus);
        minusBtn = findViewById(R.id.btnMinus);
        layout = findViewById(R.id.layoutAddTransaction);
        context = this;

        switch (activityMode) {
            case '+': {

                layout.setBackground(ContextCompat.getDrawable(context, R.drawable.gradient_background_green));
                plusBtn.setVisibility(View.GONE);
                minusBtn.setVisibility(View.GONE);

                break;
            }
            case '-': {
                layout.setBackground(ContextCompat.getDrawable(context, R.drawable.gradient_background_red));
                plusBtn.setVisibility(View.GONE);
                minusBtn.setVisibility(View.GONE);
                break;
            }

            default: {

                plusBtn.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        layout.setBackground(ContextCompat.getDrawable(context, R.drawable.gradient_background_green));
                    }
                });

                minusBtn.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        layout.setBackground((ContextCompat.getDrawable(context, R.drawable.gradient_background_red)));
                    }
                });

                break;
            }
        }

        //Toast.makeText(this, activityMode, Toast.LENGTH_SHORT).show();

        amountEdit = findViewById(R.id.editAmount);
        //categoryEdit = findViewById(R.id.editCategory);
        categorySpinner = findViewById(R.id.SpinnerCategory);
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this, R.array.SpinnerArray, android.R.layout.simple_spinner_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        categorySpinner.setAdapter(adapter);
        categorySpinner.setOnItemSelectedListener(this);

        txtDate = findViewById(R.id.txtDate);
        txtDate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Calendar cal = Calendar.getInstance();
                int year = cal.get(Calendar.YEAR);
                int month = cal.get(Calendar.MONTH);
                int day = cal.get(Calendar.DAY_OF_MONTH);

                DatePickerDialog datePickerDialog = new DatePickerDialog(
                        AddTransactionActivity.this,
                        android.R.style.Theme_Holo_Light_Dialog_MinWidth,
                        dateSetListener,
                        year, month, day);
                datePickerDialog.getWindow().setBackgroundDrawable((new ColorDrawable(Color.TRANSPARENT)));
                datePickerDialog.show();
            }
        });
        dateSetListener = new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view, int year, int month, int day) {
                month++;
                String date = day + "/" + month + "/" + year;
                txtDate.setText(date);
            }
        };

        addBtn = findViewById(R.id.btnAdd);
        addBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Double amount = Double.valueOf(amountEdit.getText().toString());
                //String category = categoryEdit.getText().toString();
                String category = categoryTxt;
                String date = txtDate.getText().toString();

                Intent intent = new Intent();
                intent.putExtra("AMOUNT", amount);
                intent.putExtra("CATEGORY", category);
                intent.putExtra("DATE", date);
                setResult(RESULT_OK, intent);
                finish();//finishing activity

            }
        });

    }

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
        categoryTxt = parent.getItemAtPosition(position).toString();
        Toast.makeText(parent.getContext(),categoryTxt,Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {

    }
}

File: app/src/main/java/com/michalbaran/incomeapp/AddTransactionActivity.kt
package com.michalbaran.incomeapp

import android.app.Activity
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat

import android.app.DatePickerDialog
import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.text.Layout
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.DatePicker
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.Spinner
import android.widget.TextView
import android.widget.Toast

import com.michalbaran.incomeapp.database.Category
import com.reactiveandroid.query.Select
import kotlinx.android.synthetic.main.activity_add_transaction.*

import java.util.Calendar

class AddTransactionActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener {


    internal var activityMode: Char = ' '
    internal var categoryTxt: String? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_transaction)

        val intent = intent
        //if (activityMode != null) {
        activityMode = intent.getCharExtra("ACTIVITY_MODE", '0')
        //}
        // activityMode = '0';


        when (activityMode) {
            '+' -> {

                layoutAddTransaction.background = ContextCompat.getDrawable(this, R.drawable.gradient_background_green)
                btnPlus.visibility = View.GONE
                btnMinus.visibility = View.GONE
            }
            '-' -> {
                layoutAddTransaction.background = ContextCompat.getDrawable(this, R.drawable.gradient_background_red)
                btnPlus.visibility = View.GONE
                btnMinus.visibility = View.GONE
            }

            else -> {

                btnPlus.setOnClickListener { layoutAddTransaction.background = ContextCompat.getDrawable(this, R.drawable.gradient_background_green) }

                btnMinus.setOnClickListener { layoutAddTransaction.background = ContextCompat.getDrawable(this, R.drawable.gradient_background_red) }
            }
        }

        //Toast.makeText(this, activityMode, Toast.LENGTH_SHORT).show();



        val categoryList = Select.from<Category>(Category::class.java).fetch()
        //ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFrom(this,categoryList, android.R.layout.simple_spinner_item);
        val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, categoryList)
        //adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        SpinnerCategory.adapter = adapter
        SpinnerCategory.onItemSelectedListener = this


        txtDate.setOnClickListener {
            val cal = Calendar.getInstance()
            val year = cal.get(Calendar.YEAR)
            val month = cal.get(Calendar.MONTH)
            val day = cal.get(Calendar.DAY_OF_MONTH)

            var dateSetListener = DatePickerDialog.OnDateSetListener { view, year, month, day ->
                var month = month
                month++
                val date = "$day/$month/$year"
                txtDate.text = date
            }

            val datePickerDialog = DatePickerDialog(
                    this@AddTransactionActivity,
                    android.R.style.Theme_Holo_Light_Dialog_MinWidth,
                    dateSetListener,
                    year, month, day)
            datePickerDialog.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
            datePickerDialog.show()
        }


        btnAdd.setOnClickListener {
            val amount = java.lang.Double.valueOf(editAmount.text.toString())
            //String category = categoryEdit).getText(.toString();
            val category = categoryTxt
            val date = txtDate.text.toString()

            val intent = Intent()
            intent.putExtra("AMOUNT", amount)
            intent.putExtra("CATEGORY", category)
            intent.putExtra("DATE", date)
            setResult(Activity.RESULT_OK, intent)
            finish()//finishing activity
        }

    }

    override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        categoryTxt = parent.getItemAtPosition(position).toString()
        Toast.makeText(this, categoryTxt, Toast.LENGTH_SHORT).show()
    }

    override fun onNothingSelected(parent: AdapterView<*>) {

    }
}
